client/rest/src/test/java/org/elasticsearch/client/RestClientMultipleHostsIntegTests.java

```diff
@@ -187,6 +202,33 @@ public void onFailure(Exception exception) {
         }
     }
 
+    /**
+     * Test host selector against a real server <strong>and</strong>
+     * test what happens after calling
+     */
+    public void testWithHostSelector() throws IOException {
+        /*
+         * note that this *doesn't* stopRandomHost(); because it might stop
+         * the first host which we use for testing the selector.
```

Comment:
I think that if the other test methods run before this one, it can still happen that they kill your host. I would restore the previous behaviour if not too complicated and cover the case where no hosts can be selected.

Reply:
That makes sense to me.

Weird that the test wasn't failing for me, but I see your reasoning. I'll adapt the test to living with the node being shut down.

Reply:
thanks